src/client/game.cpp

```diff
@@ -3960,14 +3972,15 @@ void Game::updateFrame(ProfilerGraph *graph, RunStats *stats, f32 dtime,
 	bool draw_wield_tool = (m_game_ui->m_flags.show_hud &&
 			(player->hud_flags & HUD_FLAG_WIELDITEM_VISIBLE) &&
 			(camera->getCameraMode() == CAMERA_MODE_FIRST));
+#ifndef HAVE_TOUCHSCREENGUI
 	bool draw_crosshair = (
 			(player->hud_flags & HUD_FLAG_CROSSHAIR_VISIBLE) &&
 			(camera->getCameraMode() != CAMERA_MODE_THIRD_FRONT));
-#ifdef HAVE_TOUCHSCREENGUI
-	try {
-		draw_crosshair = !g_settings->getBool("touchtarget");
-	} catch (SettingNotFoundException) {
-	}
+#else
+	bool draw_crosshair = !m_android_touchtarget ||
+			(m_android_use_crosshair &&
+			camera->getCameraMode() != CAMERA_MODE_THIRD_FRONT) ||
+			camera->getCameraMode() == CAMERA_MODE_THIRD;
```

Comment:
this totally overwrites the normal visibility calculations and is also subtly different?!
a variation of the following would be better:
```cpp
    if (m_android_touchtarget)
        draw_crosshair = false;
```
...if use_crosshair can be fit into it that is, I don't entirely get how that works

Reply:
I have rewritten it to be same as non-touch, except when in the first-person camera mode.